hermes-management/build.gradle

```
@@ -43,22 +43,29 @@ dependencies {
     testCompile group: 'org.testcontainers', name: 'kafka', version: versions.testcontainers
 }
 
+def consoleDir = "${buildDir}/generated/console"
+
 task buildHermesConsole(type: Exec) {
-    workingDir '../hermes-console'
+    def dir = '../hermes-console'
+    workingDir dir
+    outputs.dir(dir + "/dist")
+    inputs.dir(dir + "/static")
     commandLine './package.sh'
 }
 
 task attachHermesConsole(type: Copy, dependsOn: 'buildHermesConsole') {
-    from '../hermes-console/dist/static'
-    into(sourceSets.main.output.resourcesDir.path + '/static')
-}
-
-tasks.register('prepareIndexTemplate') {
+    def srcDir = '../hermes-console/dist/static'
+    def targetDir = consoleDir + '/static'
+    from srcDir
+    into targetDir
     doLast {
-        def indexPath = sourceSets.main.output.resourcesDir.path + '/static/index.html'
-        ant.move file: indexPath,
-                tofile: indexPath + '.ftl'
+        ant.move file: targetDir + '/index.html',
+            tofile: targetDir + '/index.html.ftl'
     }
 }
 
-jar.dependsOn(attachHermesConsole, 'prepareIndexTemplate')
+sourceSets {
+    main {
+        resources.srcDir consoleDir
```

you can add the task provider here to automatically wire the task dependencies to ensure they are executed in the correct order 

```suggestion
        resources.srcDir tasks.named('attachHermesConsole')
```

---

I think this will break the current behaviour of the build as the static files are not attached to the `distZip` package. This change would add a dependency that will cause static files to be included in the output.

---

ah, i see. i suggested this with the assumption that all the outputs would be added.

Do we still would need to make sure that the `ProcessResources` task dependsOn `attachHermesConsole`?

---

I think not - `attachHermesConsole` is an auxilliary task that copies static files from other subproject. It's intended to be run manually when needed.